### ISSUE TITLE
Other containers lose access to /local-media upon remounts?

### ISSUE THREAD

**Issue by Dulanic** [madslundt/docker-cloud-media-scripts#16]
I have noticieid a recent issue /w Plex and some other containers. When this container remounts the locations, the other dockers can no longer access it? Or at least plex? Is there something I can do to correct this issue?

Immediately after this docker remounts, this is the error I see from within the plex container:

root@mediaserver:/# cd data
bash: cd: data: Transport endpoint is not connected

container commands:

sudo docker create --name cloud --restart=always -v /local-media:/local-media:shared -v /local-decrypt:/local-decrypt:shared -v /docker/containers/cloud/config:/config -v /mnt/external/plexdrive:/chunks -v /var/log:/log -e CLEAR_CHUNK_MAX_SIZE="2000G" -e ENCRYPT_MEDIA="1" -e PGID=1000 -e PUID=1000 -e RCLONE_LOCAL_ENDPOINT="gd-crypt:" -e REMOVE_LOCAL_FILES_WHEN_SPACE_EXCEEDS_GB="1500" -e FREEUP_ATLEAST_GB="1000" -e TZ="America/Chicago" --privileged --cap-add=MKNOD --cap-add=SYS_ADMIN --device=/dev/fuse madslundt/cloud-media-scripts

sudo docker run -d --name plex --restart=always --network=host -e TZ="America/Chicago" -e PLEX_CLAIM="*redacted*" -v /docker/containers/plex/config:/config -v /mnt/downloads/transcode:/transcode -v /storage/Media:/data plexinc/pms-docker

**Comment by Dulanic** [comment]
Note: I attempted to use --volumes-from cloud on plex and the issue continues there. Seems like docker can't hjandle fuse remounts... I used bash in the plex contrainer and /local-decrypt is good, cloud is not.
```

root@mediaserver:/local-decrypt# cd ..
root@mediaserver:/# cd local-media
bash: cd: local-media: Transport endpoint is not connected
root@mediaserver:/# cd cloud-decrypt
bash: cd: cloud-decrypt: Transport endpoint is not connected
```

**Comment by madslundt** [comment]
Yes some people have experience the issue you address here. I'm looking into it but so far it looks like docker isn't happy about mounting folders and then sharing them with other containers.

**Comment by Dulanic** [comment]
Definitely a docker issue, if I run plex outside of the container it is fine.

**Comment by Antexa** [comment]
For now, the only thing we can do is to run plex from the host directly ?

**Comment by Dulanic** [comment]
Yup

**Comment by daniel-loader** [comment]
Doesn't this negate the portability? Right now I can either have the mounts running natively on the OS or the HTPC application stack running natively, and arguably the sonarr/plex/radarr/plexpy/ombi/hydra docker benefits outweigh this, though it'd be good to have the whole thing in a docker-compose.yml including mounting so it can be portable.

**Comment by Dulanic** [comment]
Kind of but I think it's a docker issue. Not positive but that's where madslundt was thinking.

**Comment by RaymondSchnyder** [comment]
This is definietly a docker issue, i have the same with my host based setup (apps in docker containers, plexdrive, rclone and unionfs on host).
When plexdrive, rclone or unionfs fails or restarts, all the containers cannot access the mounts anymore, a reboot of the containers one by one fixxes this, but this is annoying as hell.

**Comment by daniel-loader** [comment]
I've found that as long as the overlaying unionFS mount stays active you
can lose the underlying plexdrive/rclone mounts and remount them without
docker losing them, it's a bit of a faff if I'm honest, though I understand
why it is the way it is.

**Comment by Dulanic** [comment]
I added a docker containers restart to my mount.remote

**Comment by sea3pea0** [comment]
I was having this problem but I solved it by mounting only the root unionfs-fuse folder in the other containers. So for example, my main unionfs-fuse folder is /media-fuse. Before I was mounting /media-fuse/tv, /media-fuse/movies, etc. in other containers and if the plexdrive container went down or was restarted, the other containers would lose connection to those folders. When I started mounting /media-fuse:/data:shared in all the other containers I stopped having problems of the containers losing access to the mounts
